src/Core/OrganizationFeatures/AuthorizationHandlers/CollectionUserAuthorizationHandler.cs

```diff
@@ -0,0 +1,207 @@
+    /// </summary>
+    /// <remarks>
+    /// Checks that the following are all true:
+    /// - The target collection(s) exists 
+    /// - The acting user is an owner/admin AND the collection management setting is enabled
+    ///   OR
+    ///   The target collection(s) is manageable by the acting user. 
+    /// - The target user(s) exists and belongs to the same organization as the target collection
+    /// </remarks>
+    private async Task CanCreateAsync(AuthorizationHandlerContext context,
+        CollectionUserOperationRequirement requirement, ICollection<CollectionUser> resource)
+    {
+        if (!_currentContext.UserId.HasValue)
+        {
+            context.Fail();
+            return;
+        }
+
+        var distinctTargetCollectionIds = resource.Select(c => c.CollectionId).Distinct().ToList();
+
+        // List of collections the user is performing the operation on
+        var targetCollections = await _collectionRepository.GetManyByManyIdsAsync(distinctTargetCollectionIds);
+
+        // A target collection does not exist, fail the requirement
+        if (targetCollections.Count != distinctTargetCollectionIds.Count)
+        {
+            context.Fail();
+            return;
+        }
+
+        // TODO: Add check for future organization Collection Management setting here and skip the next check if it's enabled
+
+        // List of collections the user is allowed to manage
+        var manageableCollections = (await _collectionRepository.GetManyByUserIdAsync(_currentContext.UserId.Value)).Where(c => c.Manage).ToList();
+
+        // Any target collection is not in the list of manageable collections, fail the requirement
+        if (targetCollections.Any(tc => !manageableCollections.Exists(c => c.Id == tc.Id)))
+        {
+            context.Fail();
+            return;
+        }
+
+        var distinctTargetUserIds = resource.Select(c => c.OrganizationUserId).Distinct().ToList();
+
+        // List of users the user is performing the operation on
+        var targetUsers = await _organizationUserRepository.GetManyAsync(distinctTargetUserIds);
+
+        // A target user does not exist, fail the requirement
+        if (targetUsers.Count != distinctTargetUserIds.Count)
+        {
+            context.Fail();
+            return;
+        }
+
+        foreach (var targetUser in targetUsers)
+        {
+            var targetCollectionsForUser = from c in targetCollections
+                                           join cu in resource on c.Id equals cu.CollectionId
+                                           where cu.OrganizationUserId == targetUser.Id
+                                           select c;
+
+            // Target user is not in the same org as a collection they're being assigned to, fail the requirement
+            if (targetCollectionsForUser.Any(tc => tc.OrganizationId != targetUser.OrganizationId))
+            {
+                context.Fail();
+                return;
+            }
```

Comment:
You could incorporate this into the join, e.g.

```cs
var validCollections = from c in targetCollections
  join cu in resource on c.Id equals cu.CollectionId
  where cu.OrganizationUserId == targetUser.Id
  where c.OrganizationId == targetUser.OrganizationId
  select c;

if (validCollections.Count != resource.Count)
{
  context.Fail();
  return;
}
```

Reply:
Resolving as this is no longer relevant after removing this validation logic from the handler.